zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/health/DiskSpaceRecoveryIT.java

```diff
@@ -112,10 +112,14 @@ void shouldRecoverAfterOutOfDiskSpaceAfterExporting()
           .atMost(Duration.ofMinutes(3))
           .pollInterval(1, TimeUnit.MICROSECONDS)
           .untilAsserted(
-              () ->
-                  assertThatThrownBy(DiskSpaceRecoveryIT.this::publishMessage)
-                      .hasRootCauseMessage(
-                          "RESOURCE_EXHAUSTED: Cannot accept requests for partition 1. Broker is out of disk space"));
+              () -> {
+                // Ensure at least one snapshot with a compactable processed position before going
+                // out of disk space
+                partitionsClient.takeSnapshot();
+                assertThatThrownBy(DiskSpaceRecoveryIT.this::publishMessage)
+                    .hasRootCauseMessage(
+                        "RESOURCE_EXHAUSTED: Cannot accept requests for partition 1. Broker is out of disk space");
+              });
```

Comment:
:upside_down_face: IIRC, `takeSnapshot()` here only triggers the operation. Couldn't we just do something like:

- Publish a message (once we get a response, we know it will have been processed)
- Take a snapshot
- Await the snapshot
- Then continue with the test as usual

Right now, we're triggering a snapshot on every poll after every message. It likely works because we won't take snapshots in parallel (I hope :crossed_fingers:), and the snapshot is probably there very quickly, but I'd feel safer w.r.t. flakiness with a more structured approach.

Anyway, very minor :)

Reply:
We have to ensure a snapshot is taken with a position in segment 2 or later. If we take a snapshot once after the first command, it might not have filled the first segment. When we resume exporting, we won't be able to compact the segment and release disk space.